### PR TITLE
Update some deprecated uses of libraries

### PR DESCRIPTION
**Pull request by mdeff**

### CODE REVIEW

File: doc/tutorials/intro.rst [hunk 1]
```diff
@@ -38,14 +38,14 @@ follows.
 .. plot::
     :context: close-figs
 
-    >>> rs = np.random.RandomState(42)  # Reproducible results.
-    >>> W = rs.uniform(size=(30, 30))  # Full graph.
+    >>> rng = np.random.default_rng(1)  # Reproducible results.
```
Comment (nperraud) [inline doc/tutorials/intro.rst:41]: A big switch from the seed 42 to the seed 1. :-)

File: doc/tutorials/optimization.rst [hunk 2]
```diff
@@ -29,15 +29,15 @@ The first figure shows a plot of the original label signal, that we wish to reco
 .. plot::
     :context: close-figs
 
-    >>> rs = np.random.RandomState(42)
+    >>> rng = np.random.default_rng(42)
```
Comment (nperraud) [inline doc/tutorials/optimization.rst:32]: Shouldn't we have seed 1 here as well?

File: pygsp/filters/filter.py [hunk 3]
```diff
@@ -214,7 +214,7 @@ def filter(self, s, method='chebyshev', order=30):
         >>> import matplotlib.pyplot as plt
         >>> G = graphs.Ring(N=60)
         >>> G.estimate_lmax()
-        >>> s = np.random.RandomState(42).uniform(size=(G.N, 10))
+        >>> s = np.random.default_rng(42).uniform(size=(G.N, 10))
```
Comment (nperraud) [inline pygsp/filters/filter.py:217]: Seed 1?

File: pygsp/filters/filter.py [hunk 4]
```diff
@@ -721,7 +721,7 @@ def inverse(self):
         >>> _ = g.plot(ax=axes[0], title='original filter bank')
         >>> _ = h.plot(ax=axes[1], title='inverse filter bank')
         >>> # Filtering with the inverse reconstructs the original signal.
-        >>> x = np.random.RandomState(42).normal(size=G.N)
+        >>> x = np.random.default_rng(42).normal(size=G.N)
```
Comment (nperraud) [inline pygsp/filters/filter.py:724]: Seed 1?

File: pygsp/graphs/_layout.py [hunk 5]
```diff
@@ -79,7 +84,8 @@ def set_coordinates(self, kind='spring', **kwargs):
                     np.sin(2 * np.pi * np.arange(1, Nc + 1) / Nc))))
 
             # Coordinates of the nodes inside their communities
-            coords = np.random.rand(self.N, 2)
+            rng = np.random.default_rng(seed)
```
Comment (nperraud) [inline pygsp/graphs/_layout.py:87]: Just to check: calling default_rng with None, does not change anything right?

File: pygsp/graphs/nngraphs/nngraph.py [hunk 6]
```diff
@@ -66,7 +66,7 @@ class NNGraph(Graph):
     Examples
     --------
     >>> import matplotlib.pyplot as plt
-    >>> X = np.random.RandomState(42).uniform(size=(30, 2))
+    >>> X = np.random.default_rng(42).uniform(size=(30, 2))
```
Comment (nperraud) [inline pygsp/graphs/nngraphs/nngraph.py:69]: Here again

File: pygsp/learning.py [hunk 7]
```diff
@@ -83,8 +83,8 @@ def classification_tikhonov_simplex(G, y, M, tau=0.1, **kwargs):
 
     Construct a measurement signal from a binary mask:
 
-    >>> rs = np.random.RandomState(42)
-    >>> mask = rs.uniform(0, 1, G.n_vertices) > 0.5
+    >>> rng = np.random.default_rng(42)
```
Comment (nperraud) [inline pygsp/learning.py:86]: Here again

File: pygsp/learning.py [hunk 8]
```diff
@@ -224,8 +224,8 @@ def classification_tikhonov(G, y, M, tau=0):
...
```
Comment (nperraud) [inline pygsp/learning.py:227]: Here again

File: pygsp/learning.py [hunk 9]
```diff
@@ -293,12 +293,12 @@ def regression_tikhonov(G, y, M, tau=0):
 
     >>> filt = lambda x: 1 / (1 + 10*x)
     >>> filt = filters.Filter(G, filt)
-    >>> rs = np.random.RandomState(42)
-    >>> signal = filt.analyze(rs.normal(size=G.n_vertices))
+    >>> rng = np.random.default_rng(42)
```
Comment (nperraud) [inline pygsp/learning.py:296]: Here again

File: pygsp/tests/test_filters.py [hunk 10]
```diff
@@ -19,8 +19,8 @@ class TestCase(unittest.TestCase):
     def setUpClass(cls):
         cls._G = graphs.Sensor(123, seed=42)
         cls._G.compute_fourier_basis()
-        cls._rs = np.random.RandomState(42)
-        cls._signal = cls._rs.uniform(size=cls._G.N)
+        cls._rng = np.random.default_rng(42)
```
Comment (nperraud) [inline pygsp/tests/test_filters.py:22]: Here again, but since it is a test, it is less an issue.

**Review by nperraud — COMMENTED** [review]
Thanks a lot for your contribution.

For some reason, we have the seed 42 coming again and again. Should we uniform it toward `default_rng()` or `default_rng(1)`?

I suggest going for `default_rng()`

The rest of the PR is fine. Thanks.

**Comment by mdeff** [comment]
We want to seed the pseudo-random number generator with a seed (i.e., do `default_rng(seed)` instead of `default_rng()`) to get reproducible results (e.g., when we instantiate a random graph then check its properties).

Whether `seed` is 42, 1, or anything usually doesn't matter much. But sometimes we might prefer a random draw over another (e.g., a seed that generates a connected graph over a disconnected one).

File: doc/tutorials/intro.rst (hunk 1 above)
Comment (mdeff) [reply doc/tutorials/intro.rst:41]: That's an example where the seed 42 was giving a connected graph with `RandomState` but a disconnected one with `default_rng`. Hence I changed it to 1, which got me a connected graph.

File: pygsp/tests/test_filters.py (hunk 10 above)
Comment (mdeff) [reply pygsp/tests/test_filters.py:22]: Tests and docstrings are kinda similar, since we execute the docstrings and test their outputs.

**Comment by nperraud** [comment]
Approved. You can merge.

Thanks a lot

**Comment by mdeff** [comment]
Thank you!

**Pull request merged**